yb-voyager/src/migassessment/sizing.go

```diff
@@ -557,13 +560,14 @@ func findNumNodesNeededBasedOnTabletsRequired(sourceIndexMetadata []SourceDBMeta
 			// Iterate over each table and its indexes to find out how many tablets are needed
 			for _, table := range rec.ShardedTables {
 				_, tabletsRequired := getThresholdAndTablets(rec.NumNodes, lo.Ternary(table.Size.Valid, table.Size.Float64, 0))
-				for _, index := range sourceIndexMetadata {
-					if index.ParentTableName.Valid && (index.ParentTableName.String == (table.SchemaName + "." + table.ObjectName)) {
-						// calculating tablets required for each of the index
-						_, tabletsRequiredForIndex := getThresholdAndTablets(rec.NumNodes, lo.Ternary(index.Size.Valid, index.Size.Float64, 0))
-						// tablets required for each table is the sum of tablets required for the table and its indexes
-						tabletsRequired += tabletsRequiredForIndex
-					}
+				// Use optimized map lookup for indexes
+				tableKey := table.SchemaName + "." + table.ObjectName
+				indexes := indexLookupMap[tableKey]
```

Comment:
nit: wherever you fetch the indexes from the map for a table, you can have a check if that is present in the map or not and continue/return if it's not present 
```
indexes, found := indexLookupMap[tableKey]
if !found {
    log.Infof("table %s doesn't have any indexes", tableKey)
    continue
}
```

Reply:
this is not required as the subsequent loop takes care of it.